Tramstation

Betterwithketchup opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
Tramstation is a spiritual successor to the SS13 map of the same name. I tried to keep many of the distinctive features of the original while merging the 2 Z-levels and adjusting things to fit SS14. The main gimmick of course, is the tram running down the center. 

Tramstation has:

- A tram!
- an interior facing bridge
- a "radiation therapy room"
- a circular bar
- a large chapel

Thank you to everyone who gave suggestions and critiques. I really appreciate it.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
Variety in maps is always good, and I think this game has a need for unique map gimmicks.

## Technical details
<!-- Summary of code changes for easier review. -->
~~This map is very much WIP still.~~ Its almost done! Please provide any critique or feedback you think will be useful.
TODO:
- [x] add emergency lights
- [x] adjust meteor shielding
- [x] adjust power banks and solars
- [x] wait for https://github.com/space-wizards/space-station-14/pull/34077 or something similar for the tram
- [x] check door accesses
- [x] check firelocks and air alarms
- [x] figure out playercount and change prototype
- [x] decals and detailing everywhere
- [x] playtest and fix bugs

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
The tram!
It starts at the far left of the station
<img width="448" height="160" alt="Tram2-1" src="https://github.com/user-attachments/assets/2370d057-9b36-4bae-b1b3-0a6fe96591e6" />








Most recent render
<img width="6304" height="4000" alt="Tram2-0" src="https://github.com/user-attachments/assets/5b1bf899-b5c8-4060-98ff-abefc37edbee" />










## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added Tramstation
-->

Betterwithketchup committed:
Tram!

Betterwithketchup committed:
add tram to test file and add prototype

Betterwithketchup committed:
fixed prototype

Betterwithketchup committed:
fixed map and prototype

Aidenkrz commented:
So does like, a player literally just move a shuttle back and forth down a hallway for this?

jbox144 commented:
Tram 👍

superjj18 commented:
> So does like, a player literally just move a shuttle back and forth down a hallway for this?

Think horizontal elevator, or you know, a tram lol

Typically you just press a button to go to a particular part of the station.  And enjoy the view as you run over and hit 6 people who were impatient and got hit crossing the road :trollface:

Thatonestomf commented:
I cant wait for Z levels so that this good looking station gets even better, oh and nice thinking using a shuttle with containment fields to make it really feel like tram station, though i think that may cause a real issue in spacing and such if the tram line gets damaged in any way, and there is also the issue that there Arnt platforms from place to place between the tramline (the comedy and of course... the tram looks like it needs a driver to function and without it.... thats gonna be rough
but within the constraints you were put under, i think this is a fine enough station.

now the critiques of mine (as they come through):
you may want to reinforce the path to ai, it looks like it could get cut off really easily, like.... in under 2 mins.
Maints looks really sparse, how is a tider gonna tide without maints lockers? (maints rooms are still cool, keep up the good work).
Dorms looks depressing, give it some love.
Sci doesnt have any roundstart artifact containers. The poor poor research interns....
Chem looks really cramped.
Medical only has 1 medical vend, and its off to the side in the locker room. 2 or 3 would be nice.
Mats in general look really sparse, gonna be hard to scavenge some extras without cargo.
Brig med could use some extra meds for all the prisoners that got caught while getting hit by the tram.
Grav gen is just..... in a really strange spot, it dont even got a sign.
Only the medical sec checkpoint has a recharger. some more would be nice.
No musician rooms?
Bridge doesnt have an axe.
Det doesnt have ~~_1984_~~ a camera console.
Tools is really small and doesnt have welding fuel for welder.... torches. oh and it got no water in it.
Salv's free stuff is gone, ITS GONE ACK!
No maints bar locker?
The tech storage is command rather than engie for the first part
Boxer and zookeeper yet again denied a cool station to be apart of...., 

But dont take that as me not liking the station, i love it and cant wait to run over some poor borgs with a tram.

Qsan163 commented:
that one player who is trapped forever in pilot room of the tram to move it :

FairlySadPanda commented:
![1_6jPAb4qy1YfTgKOetmACdQ](https://github.com/user-attachments/assets/50a99221-3cdc-4f27-af74-828f159da484)

themias commented:
I really like the idea! My only concern is with the tram being a bomb-magnet.

Maybe engineering could start with spare parts to be able to rebuild it?

KashRas2 commented:
Bro still didn't realize that it wasn't a tram. It was a shuttle.

aeoli-fly commented:
It'd be nice if the command meeting room had more seats, the captain's table in their office is fine to have just 4 but the meeting room proper should have room for all heads. One could easily fit 7 seats there.

Djungelskog2 commented:
if it gets merged id love to forcemap this for admemes, id also be happy to run playtests to figure out the playercount for it

Betterwithketchup committed:
small changes

Aidenkrz commented:
> > So does like, a player literally just move a shuttle back and forth down a hallway for this?
> 
> Damn right. For anybody n' everybody.
> 
> Nukies board my tram? Tip my hat and close the door, drive straight to security. Collect my 10 spesos tip. Move on to the next station, light a cig, watch the stop behind me blow up to china lake friendly fire in the rear view mirror.
> 
> In all seriousness, a Tram Operator job would be cool, but a Service Worker will probably be fulfilling this position.

yeah but like isn't tram in 13 a grid that automatically moves between points, not just a shuttle? i feel like this would be buggy as hell, not to discredit the work the author has put in.

Betterwithketchup committed:
changed dorms and tool room

Everturning commented:
I actually think itd be better to wait for z-levels than merge this implementation

Betterwithketchup committed:
Ready for review.
please dont be too harsh i will cry

Betterwithketchup committed:
Merge branch 'master' into Tramstation

Betterwithketchup committed:
Various fixes

medical front, sci front. edited some maints. fixed cramped atmos. added warden bedroom

Betterwithketchup committed:
small updates

Betterwithketchup committed:
Merge branch 'master' into Tramstation

Betterwithketchup committed:
new maints rooms and small fixes

Betterwithketchup committed:
Merge branch 'Tramstation' of https://github.com/Betterwithketchup/space-station-14 into Tramstation

Betterwithketchup committed:
i hope this doesnt break anything

Betterwithketchup committed:
i dont think anyone reads these

FairlySadPanda commented:
`ain't no stranger to pull requests`
`you know the rules, and so do I`
`a code submission is what I'm thinking of`
`request for changes would, make you cry`
`never gonna read this code`
`never gonna engage with the comments`
`never gonna clone the branch and test it`
`I'm never gonna even try`
`just gonna say goodbye`
`gonna do my own work,`
`make a PR, too`

VerinSenpai commented:
> `ain't no stranger to pull requests` `you know the rules, and so do I` `a code submission is what I'm thinking of` `request for changes would, make you cry` `never gonna read this code` `never gonna engage with the comments` `never gonna clone the branch and test it` `I'm never gonna even try` `just gonna say goodbye` `gonna do my own work,` `make a PR, too`
> 
> :godo:

Did... Did we just get panda rolled

LaCumbiaDelCoronavirus commented:
Looks nice though a few suggestions i have:
-Add more gas miner chambers to atmos 
-Make the meteor shielding grilles outside station be closer to it, long stretches of lattice connecting the grilles to station look weird

superjj18 commented:
Oh my god I just realized the tram is just a shuttle lmao bus driver RP.

Sad that we don't get the "accidents this shift" counter and the ability to wound people

PubliclyExecutedPig commented:
hey uhm minor thing but im like 99% sure you cant walk through this **
![image](https://github.com/user-attachments/assets/1e47a273-5cb6-44ad-a87d-07fb2b596884)
**

Betterwithketchup committed:
renamed everything

Betterwithketchup committed:
Firelock 198X is a fun game

tiles under doors, fixed airlock access in parts, more maints stuff. tried to fix tram air

Betterwithketchup committed:
still no idea how many people belong on this station

Betterwithketchup committed:
Merge branch 'master' into Tramstation

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
Genpop, silos and more!

Betterwithketchup committed:
Fixed Sec and atmos pipes

Betterwithketchup committed:
i am a fool

ArtisticRoomba committed:
Merge remote-tracking branch 'upstream/master' into fork/Betterwithketchup/Tramstation

ArtisticRoomba reviewed (changes requested):
After skimming over it, my only concerns are:
- Giving engineering shuttle parts nearby to provide rapid repair
- Giving Security a way to stuff prisoners into Genpop and extract them without having to go through the Tram (I know, I know, but, feels odd).

Besides that, that's my only concerns when looking at the map at a glance. However another problem I observed with merging recent master is that shuttle collisions troll Tram significantly. However I'll see if I can get that fixed soon somehow.

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
sec entrance to genpop. more engi parts and tram adjustments

Betterwithketchup committed:
everytime i fix something i break something else

Nukesthestation commented:
mapped Hew Plusshie? Someone could make a bomb with that easily roundstart
![image](https://github.com/user-attachments/assets/6d3115c5-f6f1-416c-bfca-d22133987ef5)

walksanatora commented:
it is 1 plushie. you aint gonna get a good explosion out of that. and even if you could it is in perma so only security could and they dont have a tendency to make bombs.

PubliclyExecutedPig commented:
shuttle collision

Betterwithketchup committed:
small fixes and tram stuff

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
added turrets and attempted to fix mail system

Betterwithketchup committed:
Resolved most issues from playtest and changed prototype

Betterwithketchup committed:
Merge branch 'master' into Tramstation

Betterwithketchup committed:
small changes

Betterwithketchup committed:
Merge remote-tracking branch 'origin/Tramstation' into Tramstation

RangerXVII commented:
Maybe you could add some kind of cushion that can only be found on this map. If the wall AND the shuttle has it, it won't damage anything.

Betterwithketchup committed:
minor tweaks

Betterwithketchup committed:
More small fixes

Betterwithketchup committed:
changes from feedback

Betterwithketchup committed:
colored pipes

Betterwithketchup committed:
Merge branch 'master' into Tramstation

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
redid turrets

Betterwithketchup committed:
small fixes

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Kowlin reviewed (commented):
Some things I've discovered while running this map:

- The paramedic's locker isn't the filled kind.
- The AI turrets located at AI Upload are the wrong type, Either its turret controller will have to be replaced with the AI version, or the sentry turrets need to be replaced with the Security version.
- In one of the 2 rounds I ran, the left containment field didn't properly attach to the tram shuttle.
- The tram might need a speed limiter (Unknown if that can be done) to prevent oopsies by the tram slamming into the station 2 minutes into the round. Yes, that did happen. 
<img width="1023" height="611" alt="image" src="https://github.com/user-attachments/assets/89246ae1-4b59-4f3c-9676-778493e20aaa" />

Betterwithketchup committed:
chem, comm and other changes

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
Merge branch 'master' into Tramstation

Betterwithketchup committed:
someone else might need to finish this

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
air and power fix in sec

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
fixed vault and minor firelocks around the station

Betterwithketchup committed:
lawyer and map test/pool

Betterwithketchup committed:
bugfixing

Betterwithketchup committed:
maint and misc changes

Betterwithketchup committed:
no more diy teg

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
tram speed limited

Betterwithketchup committed:
Merge branch 'space-wizards:master' into Tramstation

Betterwithketchup committed:
changed job roster adn added Tram Driver spawn

Betterwithketchup committed:
im an idiot

Betterwithketchup committed:
final changes

Betterwithketchup committed:
fix tests

Betterwithketchup committed:
fix tests for real this time
